platforms/jvm/platform-jvm/src/main/java/org/gradle/api/plugins/JvmEcosystemPlugin.java

```diff
@@ -62,7 +62,7 @@ public void apply(Project project) {
 
     private void configureVariantDerivationStrategy(ProjectInternal project) {
         ComponentMetadataHandlerInternal metadataHandler = (ComponentMetadataHandlerInternal) project.getDependencies().getComponents();
-        metadataHandler.setVariantDerivationStrategy(JavaEcosystemVariantDerivationStrategy.getInstance());
+        metadataHandler.setVariantDerivationStrategy(objectFactory.newInstance(JavaEcosystemVariantDerivationStrategy.class));
```

Comment:
Not now, but I wonder if we should switch this to a `Class` and do the instantiate in the service.  It's all internal, but I wonder if plugins are calling this

Reply:
I think we do in some integ tests -- not sure if any external plugins are using this API.

Sounds like a good idea though